tests/conftest.py

```diff
@@ -260,40 +273,116 @@ def decam_example_raw_image( decam_example_exposure ):
-        session.add( exposure )
-        session.commit()
-        prepper = Preprocessor()
-        # NOTE: this has a side effect of loading some DECam
-        # calibration files (flats, etc.) into the databse
-        ds = prepper.run( exposure, 'N1', session=session )
-        ds.save_and_commit( session=session )
-    yield ds
-    ds.delete_everything()
 
-@pytest.fixture(scope="session")
-def decam_example_reduced_image_source_list_ds( decam_example_reduced_image_ds ):
-    """Returns the same datastore from decam_example_reduced_image_ds, only now with a source list too"""
-    det = Detector()
-    ds = det.run( decam_example_reduced_image_ds )
-    ds.save_and_commit()
-    yield ds
-    # This next line may not be necessary since
-    # decam_example_reduced_image_ds will run it
-    # ...indeed, it looks like data_store.delete_everyting()
-    # is not robust to being called more than once
-    # ds.delete_everything()
+    # This next block of code generates the files read in this test
+    #   (though they'll be written to data/115 instead of
+    #   data/test_data/DECam_examples).  Commented out here because
+    #   of course we just want to read them, but leave the commented
+    #   code here in case for whatever reason we want to regenerate them.
+    #   (In that case, run up to the pdb interrupt, and then copy the
+    #   files you want out of data/115.)
+    # with SmartSession() as session:
+    #     # Spoof md5sum so we can add it
+    #     exposure = exposure.recursive_merge( session )
+    #     session.add( exposure )
+    #     session.commit()
+    #     prepper = Preprocessor()
+    #     ds = prepper.run( exposure, 'N1', session=session )
+    #     det =  Detector()
+    #     ds = det.run( ds )
+    #     ds.save_and_commit()
+    # import pdb; pdb.set_trace()
+
+    datadir = pathlib.Path( FileOnDiskMixin.local_path )
+    filepathbase = 'test_data/DECam_examples/c4d_20221104_074232_N1_g_Sci_VWQNR2'
+
+    # The "_cached" files are the ones that are saved to the github
+    # archive.  The later call to ds.delete_everything() is going to
+    # blow away the names that match what's in the "filepath" (plus
+    # "filepath_extensions") fields of ds.image, ds.sources, and ds.psf,
+    # so we use these "cached" names as a way of keeping those files
+    # around.  Moreover, some test might modify the .fits files!
+    copiesmade = []
+    try:
+        for ext in [ '.image.fits', '.weight.fits', '.flags.fits', '.sources.fits', '.psf', '.psf.xml' ]:
+            actual = datadir / f'{filepathbase}{ext}'
+            if actual.exists():
+                raise FileExistsError( f"{actual} exists, but at this point in the tests it's not supposed to" )
+            shutil.copy2( datadir / f"{filepathbase}{ext}_cached", actual )
+            copiesmade.append( actual )
+
+        with SmartSession() as session:
+            # The filenames will not match the provenance, because the filenames
```

Comment:
No.  I put this here in case in a couple of months we're looking at these tests, to stop us from going down a rabbit hole after noticing that the provenance key in the data structures and in the filename don't match.